docs/common/lifecycle-message.md

```diff
@@ -0,0 +1,45 @@
+As implied by the name, a `#!typescript LifecycleMessage` refers to some 
+pre-defined system lifecycle, such as the flow of a commit from Git, 
+through CI to deployment and monitoring and so on.
+
+Lifecycle messages are automatically routed to appropriate chat channel by
+the Atomist Bot. Typically, this is the channel automatically created by the
+Atomist Bot when a new project is generated.
+
+Message styling is also currently performed by the Atomist Bot, though we are
+working to allow customization in the handlers themselves.
+
+The Atomist Bot needs two pieces of information in order to correctly route and
+render a lifecycle message, a [GraphNode](/user-guide/rug/path-expressions) and
+a `lifecycle-id`. The GraphNode contains all the relevant data associated with
+an event required to render actionable messages in chat, and the lifecycle-id
+ties associates related messages to one-another. In a Slack context, this means
+a Commit message could have its associated CI build status updated as the build
+progresses from say `started` through `succeeded` or `failed`.
+
+So the main responsibility of lifecycle messages is to create associations between
+related messages using the lifecycle-id.
+
+Mesages would not be actionable without some way for the user to initiate related
+acitivites. Lifecyce messages achieve this by allowing `actions` in the form of a
+`#!typescript Presentable` to be added to the message before dispatch:
+
+```typescript
+ message.addAction({
+            label: "Reopen",
+            instruction: {
+                kind: "command",
+                name: "ReopenGitHubIssue",
+                parameters: {
+                    issue: issue.number,
+                    owner: issue.repo.owner,
+                    repo: issue.repo.name,
+                },
+            },
+        });
+```
+
+Here we are saying: render a button with label `Reopen`, which when clicked, will
+dispatch a [Command Handler](/user-guide/rug/command-handlers) (`#!typescript "command"`)
+called `ReopenGithubIssue`, which will execute a [Rug Function](/user-guide/rug/rug-functions.md)
+which attempt to re-open the issue.
```

Comment:
Well - this whole area is nuts IMO and is going to be changed when (pretty soon I think) move all the rendering in to the rugs themselves.